zcash_client_sqlite/src/wallet/db.rs

```diff
@@ -497,219 +518,201 @@ pub(super) const TABLE_SQLITE_SEQUENCE: &str = "CREATE TABLE sqlite_sequence(nam
+    blocks_max_height AS (
+        SELECT MAX(blocks.height) AS max_height FROM blocks
+    )
+SELECT notes.account_id             AS account_id,
+       notes.block                  AS mined_height,
+       notes.txid                   AS txid,
+       transactions.tx_index        AS tx_index,
+       transactions.expiry_height   AS expiry_height,
+       transactions.raw             AS raw,
+       SUM(notes.value)             AS account_balance_delta,
+       transactions.fee             AS fee_paid,
+       SUM(notes.is_change) > 0     AS has_change,
+       MAX(COALESCE(sent_note_counts.sent_notes, 0))  AS sent_note_count,
+       SUM(notes.received_count)         AS received_note_count,
+       SUM(notes.memo_present) + MAX(COALESCE(sent_note_counts.memo_count, 0)) AS memo_count,
+       blocks.time                       AS block_time,
+       (
             blocks.height IS NULL
             AND transactions.expiry_height BETWEEN 1 AND blocks_max_height.max_height
-        ) AS expired_unmined
+       ) AS expired_unmined
 FROM notes
 LEFT JOIN transactions
-        ON notes.txid = transactions.txid
+     ON notes.txid = transactions.txid
 JOIN blocks_max_height
 LEFT JOIN blocks ON blocks.height = notes.block
 LEFT JOIN sent_note_counts
-        ON sent_note_counts.account_id = notes.account_id
-        AND sent_note_counts.txid = notes.txid
+     ON sent_note_counts.account_id = notes.account_id
+     AND sent_note_counts.txid = notes.txid
 GROUP BY notes.account_id, notes.txid";
 
 pub(super) const VIEW_TX_OUTPUTS: &str = "
 CREATE VIEW v_tx_outputs AS
-SELECT transactions.txid              AS txid,
-        v_received_notes.pool          AS output_pool,
-        v_received_notes.output_index  AS output_index,
-        sent_notes.from_account_id     AS from_account_id,
-        v_received_notes.account_id    AS to_account_id,
-        NULL                           AS to_address,
-        v_received_notes.value         AS value,
-        v_received_notes.is_change     AS is_change,
-        v_received_notes.memo          AS memo
-FROM v_received_notes
+SELECT transactions.txid            AS txid,
+       ro.pool                      AS output_pool,
+       ro.output_index              AS output_index,
+       sent_notes.from_account_id   AS from_account_id,
+       ro.account_id                AS to_account_id,
+       NULL                         AS to_address,
+       ro.value                     AS value,
+       ro.is_change                 AS is_change,
+       ro.memo                      AS memo
+FROM v_received_outputs ro
 JOIN transactions
-    ON transactions.id_tx = v_received_notes.tx
+    ON transactions.id_tx = ro.transaction_id
 LEFT JOIN sent_notes
-    ON sent_notes.id = v_received_notes.sent_note_id
-UNION
-SELECT utxos.prevout_txid           AS txid,
-        0                            AS output_pool,
-        utxos.prevout_idx            AS output_index,
-        NULL                         AS from_account_id,
```

Comment:
AFAICT (at least as of the first commit in the PR), this `NULL AS from_account_id` is preserved, because there is still an asymmetry between transparent and shielded (we are not inserting transactions into both sent and received tables for transparent, in the same way we do for shielded). I think that asymmetry needs fixing prior to the next release (if not in this PR).

Reply:
Specifically, there is an asymmetry between transparent and shielded in the following places:
- `WalletDb::put_blocks`
  - Likely non-blocking: this does nothing with transparent details, but that makes sense for now as we don't currently have transparent logic in `scan_cached_blocks`. However, we do need to ensure that any transparent aspects influenced by block scanning do get updated (the one that comes to mind is `transactions.block` which I think _is_ correctly handled in this PR).
- `WalletDb::store_decrypted_tx`
  - Calls both `put_received_note` and `put_sent_output` for both Sapling and Orchard depending on detection code, but only ever calls `put_sent_output` for transparent.
- `WalletDb::store_decrypted_tx`
  - Calls `put_received_note` for `sent_tx.outputs()` that are shielded, but not for transparent outputs.

Reply:
Opened https://github.com/zcash/librustzcash/issues/1434